src/ConstructiveSolidGeometry/VolumePrimitives/Sphere.jl

```diff
@@ -31,13 +30,19 @@ end
 
 get_r_limits(s::Sphere{T}) where {T} = (_left_radial_interval(s.r), _right_radial_interval(s.r))
 
+get_decomposed_surfaces(s::Sphere{T, T}) where {T} = AbstractSurfacePrimitive[SphereMantle{T}(s.r)]
+
+function get_decomposed_surfaces(s::Sphere{T, <:AbstractInterval{T}}) where {T}
+    rMin::T, rMax::T = get_r_limits(s)
+    AbstractSurfacePrimitive[SphereMantle{T}(rMin), SphereMantle{T}(rMax)]
```

Comment:
Can this be a tuple?
At least remove the `AbstractSurfacePrimitive` in front of the vector.

Reply:
Same at line 33

Reply:
This is for consistency. `get_decomposed_surfaces` always returns an `AbstractSurfacePrimitive` vector for all volume primitives. The Sphere is just special in the since that the only type of decomposed surfaces it has are `SphereMantle`

Reply:
This creates type instabilities.

I know we merge these arrays so they become arrays of multiple types (what we have to address at some point),
but here we should definitely not return an array of abstract element type here.

Reply:
We save decomposed surfaces to `Semiconductor`, `Contact`, and `Passive`. Does it make sense to store them as `Tuple` instead of as `Vector` (to save the type information)?

Reply:
`get_decomposed_surfaces` is never type stable. I think eventually we will need a better fix, but for now this is what I have been working with

Reply:
I suggest we leave this PR as is and resolve the issue as a whole in a separate PR. It would be a change to all `get_decomposed_surfaces` of every Volume Primitive

Reply:
That sounds reasonable.

Reply:
Well, the method `get_decomposed_surfaces(s::Sphere{T, <:AbstractInterval{T}})` clearly always returns two `SphereMantle{T}` so this is always type stable. 

But if this also the case for the other primitives right now we can leave it like it is for now and clean this up later together with the other primitives. Just removing the `AbstractSurfacePrimitive` in front of the `[` should not break anything anyhow.

Reply:
This happens in other volume primitives too, not like here where this is always the case but in specific scenarios.